src/components/PollSection/usePoll.ts

```diff
@@ -0,0 +1,59 @@
+import { useEffect, useMemo, useState } from 'react';
+import dayjs from 'dayjs';
+import { PollRepository } from '@amityco/ts-sdk-react-native';
+
+export const usePoll = (pollId: string, shouldFetch: boolean) => {
+  const [pollData, setPollData] = useState<Amity.Poll | undefined>(undefined);
+
+  const endDays = useMemo(() => {
+    if (pollData?.closedAt) {
+      const endDate = dayjs(pollData.closedAt);
+      const daysRemaining = endDate.diff(dayjs(), 'day');
+      return daysRemaining || 0;
+    }
+    return 0;
+  }, [pollData]);
+
+  const isPollClosed = useMemo(() => {
+    if (pollData?.status) {
+      return pollData.status === 'closed';
+    }
+    return false;
+  }, [pollData]);
+
+  const totalVote = useMemo(() => {
+    if (pollData?.answers?.length > 0) {
+      const total = pollData.answers.reduce((acc, poll) => {
+        return acc + poll.voteCount;
+      }, 0);
+      return total;
+    } else {
+      return 0;
+    }
+  }, [pollData]);
+
+  const isAlreadyVoted = useMemo(() => {
+    if (pollData?.answers) {
+      return pollData.answers.some((answer) => answer.isVotedByUser);
+    }
+    return false;
+  }, [pollData]);
```

Comment:
we should extract this to another utils function or hooks because this hook will getting bigger and bigger

Reply:
This file has only 57 lines and we just return 4 values. These values are memoized and depends on pollData from useEffect return. 
I think we don't need to extract for now. May be later we can do that.